<launch>
<!-- standalone launch -->
  <!-- <include file="$(find reemc_gazebo)/launch/reemc_empty_world.launch" /> -->


  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find reemc_wbc_tutorials)/config/rviz/reemc_wbc.rviz" /> 

  <!-- Upload capsule collision operations  & description-->
  <include file="$(find capsule_collision)/launch/reemc_capsule_description.launch" />

  <!-- Configure wbc -->
  <rosparam command="load"
    file="$(find reemc_wbc_tutorials)/config/reemc_wbc_humanoids_ft_imu_right_arm_only.yaml" />
  
  <!-- Floating base transform publisher -->
  <node name="floating_base_publisher"
        pkg="pal_wbc_controller" type="floating_base_publisher" />

  <!-- Spawn controller -->
  <node name="wbc_spawner"
        pkg="controller_manager" type="spawner" output="screen"
        args="whole_body_kinematic_controler" />


    <!-- Node to follow a goal but allow disturbances on the end effector -->
    <node pkg="move_by_ft_wrist" type="arm_by_ft_wrist_hardcoded_frame.py" name="arm_by_ft_wrist" output="screen">
        <rosparam command="load" file="$(find move_by_ft_wrist)/config/armby.yaml"/>
        <remap from="/right_wrist_ft" to="/right_wrist_ft_fake"/>
    </node>

    <node pkg="move_by_ft_wrist" type="force_torque_publisher.py" name="force_torque_publisher" output="screen">
<!--         <remap from="/right_wrist_ft_fake" to="/right_wrist_ft_fake"/> -->
    </node>

    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconf_stuff" />

    <node pkg="move_by_ft_wrist" type="interactive_marker_posestamped.py" name="im_posestamped">
        <remap from="/whole_body_kinematic_controler/arm_right_7_link_goal" to="/pose_to_follow" />
    </node>

    <node pkg="move_by_ft_wrist" type="admitance_handshake.py" name="admitance_handshake" output="screen">
      <remap from="/right_wrist_ft" to="/right_wrist_ft_fake"/>
    </node>

    <node pkg="move_by_ft_wrist" type="close_hand_by_ft.py" name="close_hand_by_ft" output="screen">
      <remap from="/right_wrist_ft" to="/right_wrist_ft_fake"/>
    </node>  

  <include file="$(find reemc_controller_configuration)/launch/hey5_controller.launch"/>

</launch>